### PR TITLE
Switch from assert to deeper

### PR DESCRIPTION
**Pull request by leplatrem**
Use deeper instead of assert, which is shipped entirely by browserify.
(as an example, it reduces the gecko build by 25Ko)

I had to blacklist `buffertools` because of this: https://github.com/othiym23/node-deeper/pull/4

@n1k0 r?

### CODE REVIEW

**Comment by n1k0** [comment]
Test failures are worrying...

File: package.json [hunk 1]
```diff
@@ -8,10 +8,10 @@
...
+    "dist-dev": "browserify -s Kinto -d --ignore buffertools -e src/index.js -o dist/kinto-$npm_package_version.js -t [ babelify --sourceMapRelative . ]",
+    "dist-noshim": "browserify -s Kinto -g uglifyify --ignore buffertools --ignore isomorphic-fetch --ignore babel-polyfill -e src/index.js -o dist/kinto-$npm_package_version.noshim.js -t [ babelify --sourceMapRelative . ]",
+    "dist-prod": "browserify -s Kinto -g uglifyify --ignore buffertools -e src/index.js -o dist/kinto-$npm_package_version.min.js -t [ babelify --sourceMapRelative . ]",
+    "dist-fx": "BABEL_ENV=firefox browserify -s loadKinto --ignore buffertools --ignore isomorphic-fetch -e fx-src/index.js -o temp.jsm -t [ babelify --sourceMapRelative . ] && mkdir -p dist && cp fx-src/jsm_prefix.js dist/moz-kinto-client.js && echo \"\n/*\n * Version $npm_package_version - $(git rev-parse --short HEAD)\n */\n\" >> dist/moz-kinto-client.js && cat temp.jsm >> dist/moz-kinto-client.js && rm temp.jsm",
```
Comment (n1k0) [inline package.json]: At some point we should investigate if browserify common options could be configured in JSON in a dedicated section of this file...

File: src/utils.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 "use strict";
 
-import { deepEqual } from "assert";
+import { deepEqual } from "deeper";
```
Comment (n1k0) [inline src/utils.js]: I think it's `import deepEqual from "deeper"` ;)

**Comment by leplatrem** [comment]
> Test failures are worrying...

My fault. Contrary to assert it does not throw if false :)

File: package.json (hunk 1 above)
Comment (leplatrem) [inline package.json]: could not find any doc about that

Comment (n1k0) [inline package.json]: that's sad... oh well

**Comment by n1k0** [comment]
r+

**Pull request merged**